src/Orleans.Runtime/Placement/ResourceOptimizedPlacementDirector.cs

```diff
@@ -0,0 +1,257 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+using Orleans.Runtime.Configuration.Options;
+
+namespace Orleans.Runtime.Placement;
+
+internal sealed class ResourceOptimizedPlacementDirector : IPlacementDirector, ISiloStatisticsChangeListener
```

Comment:
You could add a link to your blog post on the topic here

Reply:
I've linked it below at the `DualModeKalmanFilter<T>` class, but I can bring it to the top np